src/validation.ts

```diff
@@ -240,6 +281,11 @@ export function array<A>(validator: Validator<A>): Validator<A[]> {
 
     return errors.length ? Err(errors) : Ok(validatedArray)
   })
+
+  ;(arrayValidator as any).__tag = 'array'
+  ;(arrayValidator as any).__value = validator
```

Comment:
`value` is a bit generic; maybe `itemValidator` ? Also, since object `props` was kind of the first meta around, what do you think about consolidating? Perhaps props could be inside the meta object? `props` and this new `__value` have a lot in common. Or perhaps rename `props` and `union` to `value` ?.

Reply:
Rename `__value`, `props`, `union` as a `meta.something` ?

Reply:
yeah what do you think? Although the __ convention also works but it feels nicer to be able to just copy the entire meta object in one go.

So: 

meta.value
meta.props
meta.union

? We can also keep props at the top level as well for backward compatibility maybe.